build.js

```diff
@@ -728,57 +661,108 @@ function copyResources() {
         'sourceMappingURL=core.js.map'
     );
 
-    fs.writeFile(corejsPath, result, 'utf8', function(err) {
+    fs.writeFileSync(corejsPath, result, 'utf8', (err) => {
       if (err) return console.log(err);
     });
   });
-}
+};
 
-async function build(bcd, webref) {
+const buildManifest = (tests) => {
   const manifest = {
-    items: [],
-    individualItems: {}
+    main: {},
+    individual: {}
   };
 
-  loadCustomTests();
-  for (const buildFunc of [buildCSS, buildIDL]) {
-    const [items, individualItems] = buildFunc(bcd, webref);
-    for (let [protocol, pathname] of items) {
-      if (!pathname.startsWith('/')) {
-        pathname = `/${pathname}`;
+  for (const [ident, test] of Object.entries(tests)) {
+    for (const scope of test.scope) {
+      let endpoint = '';
+      switch (scope) {
+        case 'Window':
+          endpoint = '/api/interfaces';
+          break;
+        case 'Worker':
+        case 'DedicatedWorker':
+          endpoint = '/api/workerinterfaces';
+          break;
+        case 'ServiceWorker':
+          endpoint = '/api/serviceworkerinterfaces';
+          break;
+        case 'CSS':
+          endpoint = '/css/properties';
+          break;
+      }
+
+      if (endpoint) {
+        if (!(endpoint in manifest.main)) {
+          manifest.main[endpoint] = {
+            scope: scope,
+            httpsOnly: scope === 'ServiceWorker',
+            entries: []
+          };
+        }
+        if (!(ident in manifest.main[endpoint].entries)) {
+          manifest.main[endpoint].entries.push(ident);
+        }
       }
-      manifest.items.push({pathname, protocol});
     }
-    if (individualItems) {
-      for (const item of individualItems) {
-        manifest.individualItems[item] = item.replace(/\./g, '/');
+
+    let url = '';
+    for (const part of ident.split('.')) {
+      url += '/' + part;
+
+      if (['/api', '/css', '/css/properties'].includes(url)) {
+        // Ignore things tested in main endpoints
+        continue;
+      }
+
+      if (!(url in manifest.individual)) {
+        manifest.individual[url] = [];
+      }
+      if (!(ident in manifest.individual[url])) {
+        manifest.individual[url].push(ident);
       }
     }
   }
-  await writeManifest(manifest);
-  copyResources();
-}
+
+  return manifest;
+};
+
+/* istanbul ignore next */
+const build = async (webref, bcd) => {
+  const IDLTests = buildIDL(webref);
+  const CSSTests = buildCSS(webref, bcd);
+  const tests = Object.assign(IDLTests, CSSTests);
```

Comment:
```suggestion
  const tests = Object.assign({}, IDLTests, CSSTests);
```

To avoid modifying `IDLTests` in a way that's not obvious.